stacker/actions/build.py

```diff
@@ -192,18 +193,37 @@ def _launch_stack(self, stack, **kwargs):
             provider_stack = None
 
         old_status = kwargs.get("status")
-        if provider_stack and old_status == SUBMITTED:
-            logger.debug(
-                "Stack %s provider status: %s",
-                stack.fqn,
-                self.provider.get_stack_status(provider_stack),
-            )
-            if self.provider.is_stack_completed(provider_stack):
-                submit_reason = getattr(old_status, "reason", None)
-                return CompleteStatus(submit_reason)
-            elif self.provider.is_stack_in_progress(provider_stack):
-                logger.debug("Stack %s in progress.", stack.fqn)
-                return old_status
+
+        logger.debug("Stack %s old_status=%r  provider=%r",
+                     stack.fqn, old_status, provider_stack)
+
+        if provider_stack:
+            if old_status == SUBMITTED:
+                logger.debug(
+                    "Stack %s provider status: %s",
+                    stack.fqn,
+                    self.provider.get_stack_status(provider_stack),
+                )
+                if self.provider.is_stack_completed(provider_stack):
+                    submit_reason = getattr(old_status, "reason", None)
+                    return CompleteStatus(submit_reason)
+                elif self.provider.is_stack_in_progress(provider_stack):
+                    logger.debug("Stack %s in progress.", stack.fqn)
+                    return old_status
+                elif self.provider.is_stack_rollback(provider_stack):
+                    logger.debug("Stack %s is rolling back", stack.fqn)
+                    return CompleteStatus("Failed to create/update stack")
+            elif old_status == PENDING:
+                if self.provider.is_stack_rollback(provider_stack):
+                    logger.info("Stack %s in rollback - can't start",
+                                stack.fqn)
+                    return CompleteStatus("Failed to create/update stack")
+                elif self.provider.is_stack_rollback_complete(provider_stack):
+                    logger.info("Stack %s failed to create, deleting",
+                                stack.fqn)
+                    self.provider.destroy_stack(provider_stack)
```

Comment:
It's extra strange here because since the `old_status` is in `PENDING`, that means that this instantiation of `stacker` hasn't yet tried to execute the stack - so some other thing must have tried to launch that stack, and it seems dangerous to have 2 tools working on this at the same time.